### ISSUE TITLE
NullReferenceException when disabling GazeProvider

### ISSUE THREAD

**Issue by gilbdev** [microsoft/MixedRealityToolkit-Unity#4359]
## Describe the bug

```
NullReferenceException: Object reference not set to an instance of an object
Microsoft.MixedReality.Toolkit.Input.BaseGenericInputSource..ctor (System.String name, Microsoft.MixedReality.Toolkit.Input.IMixedRealityPointer[] pointers, Microsoft.MixedReality.Toolkit.Input.InputSourceType sourceType) (at Assets/MixedRealityToolkit/Providers/BaseGenericInputSource.cs:23)
Microsoft.MixedReality.Toolkit.Input.GazeProvider.get_GazeInputSource () (at Assets/MixedRealityToolkit.Services/InputSystem/GazeProvider.cs:106)
Microsoft.MixedReality.Toolkit.Input.GazeProvider.OnDisable () (at Assets/MixedRealityToolkit.Services/InputSystem/GazeProvider.cs:431)

```
when OnDisable is called, it's calling `InputSystem?.RaiseSourceLost(GazeInputSource);` which access the get property for GazeInputSource which tries to recreate the gazeInputSource.
(and in BaseGenericInputSource, constructor access InputSystem,that may have been destryoed before)

=> simple ugly solution could jsut be to test if InputSystem is not null MixedRealityToolkit.InputSystem?.GenerateNewSourceId()
=> a better solution would be to not recreate gaze if this is during destroy/disable process

## To reproduce

- play in editor
- stop 
may have some other setup, but nothing in particular that i can think of
[EDIT] may be related to multiscene loading with single option :p

## Expected behavior

no exception when disabling (mostly quitting the app)

## Your Setup (please complete the following information)

- Unity Version 2018.3.11f1
- MRTK Version v2.0 RC1 refresh

## Target Platform (please complete the following information)

- Unity Editor
- HoloLens

## additional info
may be related to #2744

**Comment by gilbdev** [comment]
GazeProvider.InputSystem is set by MixedRealityInputSystem class
but when MixedRealityInputSystem is disabled, GazeProvider.InputSystem is not changed

adding
`GazeProvider.InputSystem = null;`
 just before
`GazeProvider = null;`
in MixedRealityInputSystem.cs Disable() function fixes the exception (as InputSystem?.RaiseSourceLost(GazeInputSource) won't be called)
not sure this is the cleanest way, so i'l leave it to you ;), but i guess you have a good overview of the problem now ^^

**Comment by gilbdev** [comment]
fixed by @keveleigh #4498 #4222
